src/Servers/Kestrel/test/InMemory.FunctionalTests/ResponseHeaderTests.cs

```diff
@@ -100,4 +101,42 @@ public async Task ResponseHeaders_WithInvalidValuesAndCustomEncoder_AbortsConnec
 
         await connection.ReceiveEnd();
     }
+
+    [Fact]
+    public async Task ResponseHeaders_NullEntriesAreIgnored()
+    {
+        string tag = "tag";
+
+        await using var server = new TestServer(context =>
+        {
+            Assert.Empty(context.Response.Headers[tag]);
+            var beforeLength = context.Response.Headers.ContentLength;
```

Comment:
ContentLength is actually one of the HTTP headers, not the length of the collection. Remove this check.